File: AGENDLY_APP/lib/features/staff/infrastructure/repositories/staff_repository_impl.dart
import 'package:agendly_app/features/staff/domain/datasources/staff_datasource.dart';
import 'package:agendly_app/features/staff/domain/entities/staff.dart';
import 'package:agendly_app/features/staff/domain/repositories/staff_repository.dart';

class StaffRepositoryImpl extends StaffRepository {
  final StaffDatasource datasource;

  StaffRepositoryImpl(this.datasource);

  @override
  Future<List<Staff>> getAll() {
    return datasource.getAll();
  }

  @override
  Future<Staff> getById(String id) {
    return datasource.getById(id);
  }

  @override
  Future<Staff> create(Map<String, dynamic> data) {
    return datasource.create(data);
  }

  @override
  Future<Staff> update(String id, Map<String, dynamic> data) {
    return datasource.update(id, data);
  }

  @override
  Future<void> delete(String id) {
    return datasource.delete(id);
  }
}

File: BACK/src/infrastructure/repositories/staff.repository.impl.ts
import { StaffRepository } from '../../domain/repositories/staff.repository';
import { StaffEntity } from '../../domain/entities/staff.entity';
import { StaffDatasource } from '../../domain/datasources/staff.datasource';
import { StaffMapper } from '../mappers/staff.mapper';

export class StaffRepositoryImpl implements StaffRepository {
  constructor(private readonly datasource: StaffDatasource) {}

  async create(staff: Omit<StaffEntity, 'id' | 'createdAt' | 'updatedAt'>): Promise<StaffEntity> {
    const doc = await this.datasource.create(staff);
    return StaffMapper.fromMongo(doc);
  }

  async findByBusinessId(businessId: string): Promise<StaffEntity[]> {
    const docs = await this.datasource.findByBusinessId(businessId);
    return docs.map((doc) => StaffMapper.fromMongo(doc));
  }

  async findById(id: string, businessId: string): Promise<StaffEntity | null> {
    const doc = await this.datasource.findById(id, businessId);
    return doc ? StaffMapper.fromMongo(doc) : null;
  }

  async update(id: string, businessId: string, data: Partial<StaffEntity>): Promise<StaffEntity> {
    const doc = await this.datasource.update(id, businessId, data);
    return StaffMapper.fromMongo(doc);
  }
}
